### PR TITLE
Core/Misc: More coding standards

### PR DESCRIPTION
**Pull request by Aokromes**
**Changes proposed:**

-  remove more "  "
-  align things where they are aligned and it's clear it's intended
-  

**Target branch(es):** 3.3.5/master

- [x] 3.3.5

**Issues addressed:** Closes #  (insert issue tracker number)

**Tests performed:** (Does it build, tested in-game, etc.)

**Known issues and TODO list:** (add/remove lines as needed)

- [ ] 
- [ ]

### CODE REVIEW

**Comment by Shauren** [comment]
You mean more of the git blame headache

File: src/server/game/Battlegrounds/ArenaTeam.h [hunk 1]
```diff
@@ -173,10 +173,10 @@ class TC_GAME_API ArenaTeam
         int32  GetRatingMod(uint32 ownRating, uint32 opponentRating, bool won);
         float  GetChanceAgainst(uint32 ownRating, uint32 opponentRating);
         int32  WonAgainst(uint32 Own_MMRating, uint32 Opponent_MMRating, int32& rating_change);
```
Comment (jackpoz) [inline src/server/game/Battlegrounds/ArenaTeam.h]: I would fix "int32  " and "float  " and whatever "return_type  " is there too while at it

File: src/server/game/Groups/Group.h [hunk 2]
```diff
@@ -137,15 +137,15 @@ class Roll : public LootValidatorRef
         uint32 itemid;
         int32  itemRandomPropId;
         uint32 itemRandomSuffix;
-        uint8 itemCount;
+        uint8  itemCount;
```
Comment (jackpoz) [inline src/server/game/Groups/Group.h]: don't add new spaces

File: src/server/game/Groups/Group.h [hunk 3]
```diff
@@ -185,24 +185,24 @@ class TC_GAME_API Group
...
+        void LoadGroupFromDB(Field* field);
+        void LoadMemberFromDB(ObjectGuid::LowType guidLow, uint8 memberFlags, uint8 subgroup, uint8 roles);
+        bool AddInvite(Player* player);
+        void RemoveInvite(Player* player);
+        void RemoveAllInvites();
+        bool AddLeaderInvite(Player* player);
+        bool AddMember(Player* player);
+        bool RemoveMember(ObjectGuid guid, RemoveMethod const& method = GROUP_REMOVEMETHOD_DEFAULT, ObjectGuid kicker = ObjectGuid::Empty, char const* reason = nullptr);
+        void ChangeLeader(ObjectGuid guid);
+ static void ConvertLeaderInstancesToGroup(Player* player, Group* group, bool switchLeader);
```
Comment (jackpoz) [inline src/server/game/Groups/Group.h]: move static aligned to other rows void

File: src/server/game/Guilds/Guild.h [hunk 4]
```diff
@@ -254,9 +254,9 @@ class GuildBankRightsAndSlots
         void SetSlots(uint32 _slots) { slots = _slots; }
         void SetRights(uint8 _rights) { rights = _rights; }
 
-        int8 GetTabId() const { return tabId; }
+        int8  GetTabId() const { return tabId; }
         int32 GetSlots() const { return slots; }
-        int8 GetRights() const { return rights; }
+        int8  GetRights() const { return rights; }
```
Comment (jackpoz) [inline src/server/game/Guilds/Guild.h]: don't add new spaces

File: src/server/game/Loot/LootMgr.h [hunk 5]
```diff
@@ -34,15 +34,15 @@ struct LootItem;
...
-    bool    needs_quest;                                    // quest drop (quest is required for item to drop)
-    uint8   groupid;
-    uint8   mincount;                                       // mincount for drop items
-    uint8   maxcount;                                       // max drop count for the item mincount or Ref multiplicator
-    ConditionContainer conditions;                               // additional loot condition
+    uint32 itemid;                                         // id of the item
+    uint32 reference;                                      // referenced TemplateleId
+    float  chance;                                         // chance to drop for both quest and non-quest items, chance to be used for refs
+    uint16 lootmode;
+    bool   needs_quest;                                    // quest drop (quest is required for item to drop)
```
Comment (jackpoz) [inline src/server/game/Loot/LootMgr.h]: remove spaces here too

File: src/server/game/Movement/Spline/MoveSplineInitArgs.h [hunk 6]
```diff
@@ -33,8 +33,8 @@ namespace Movement
         struct {
             float x, y, z;
         } f;
-        uint64  target;
-        float   angle;
+        uint64 target;
+        float  angle;
```
Comment (jackpoz) [inline src/server/game/Movement/Spline/MoveSplineInitArgs.h]: remove spaces here too

File: src/server/game/Movement/Spline/MoveSplineInitArgs.h [hunk 7]
```diff
@@ -50,21 +50,21 @@ namespace Movement
         PointsArray path;
         FacingInfo facing;
         MoveSplineFlag flags;
-        int32 path_Idx_offset;
-        float velocity;
-        float parabolic_amplitude;
-        float time_perc;
+        int32  path_Idx_offset;
```
Comment (jackpoz) [inline src/server/game/Movement/Spline/MoveSplineInitArgs.h]: don't add new spaces

**Comment by jackpoz** [comment]
This PR adds more spaces than the ones it removes. Please just remove them all, we are not building a newspaper, unaligned code is alright.

File: src/server/game/Groups/GroupMgr.h [hunk 8]
```diff
@@ -41,6 +41,8 @@ class TC_GAME_API GroupMgr
     Group* GetGroupByDbStoreId(uint32 storageId) const;
     void   SetGroupDbStoreSize(uint32 newSize) { GroupDbStore.resize(newSize); }
 
+    void Update(uint32 diff);
```
Comment (jackpoz) [inline src/server/game/Groups/GroupMgr.h]: where did this come from ?

Comment (Aokromes) [reply src/server/game/Groups/GroupMgr.h]: update local branch with github xd

File: src/server/game/Entities/Unit/Unit.h [hunk 9]
```diff
@@ -1483,18 +1483,18 @@ class TC_GAME_API Unit : public WorldObject
 
         void ModifyAuraState(AuraStateType flag, bool apply);
         uint32 BuildAuraStateUpdateForTarget(Unit* target) const;
-        bool HasAuraState(AuraStateType flag, SpellInfo const* spellProto = nullptr, Unit const* Caster = nullptr) const;
+        bool   HasAuraState(AuraStateType flag, SpellInfo const* spellProto = nullptr, Unit const* Caster = nullptr) const;
```
Comment (jackpoz) [inline src/server/game/Entities/Unit/Unit.h]: remove the new space

File: src/server/game/Entities/Unit/Unit.h [hunk 10]
```diff
@@ -1483,18 +1483,18 @@ class TC_GAME_API Unit : public WorldObject
...
         void UnsummonAllTotems();
-        bool IsMagnet() const;
-        Unit* GetMeleeHitRedirectTarget(Unit* victim, SpellInfo const* spellInfo = nullptr);
+        bool   IsMagnet() const;
```
Comment (jackpoz) [inline src/server/game/Entities/Unit/Unit.h]: remove the new space

File: src/server/game/Entities/Unit/Unit.h [hunk 11]
```diff
@@ -1483,18 +1483,18 @@ class TC_GAME_API Unit : public WorldObject
...
+        Unit*  GetMeleeHitRedirectTarget(Unit* victim, SpellInfo const* spellInfo = nullptr);
```
Comment (jackpoz) [inline src/server/game/Entities/Unit/Unit.h]: remove the new space

File: src/server/game/Entities/Unit/Unit.h [hunk 12]
```diff
@@ -1483,18 +1483,18 @@ class TC_GAME_API Unit : public WorldObject
...
 
         int32  SpellBaseDamageBonusDone(SpellSchoolMask schoolMask) const;
         uint32 SpellDamageBonusDone(Unit* victim, SpellInfo const* spellProto, uint32 pdamage, DamageEffectType damagetype, uint8 effIndex, Optional<float> const& donePctTotal, uint32 stack = 1) const;
         float  SpellDamagePctDone(Unit* victim, SpellInfo const* spellProto, DamageEffectType damagetype) const;
         uint32 SpellDamageBonusTaken(Unit* caster, SpellInfo const* spellProto, uint32 pdamage, DamageEffectType damagetype) const;
         int32  SpellBaseHealingBonusDone(SpellSchoolMask schoolMask) const;
         uint32 SpellHealingBonusDone(Unit* victim, SpellInfo const* spellProto, uint32 healamount, DamageEffectType damagetype, uint8 effIndex, Optional<float> const& donePctTotal, uint32 stack = 1) const;
-        float SpellHealingPctDone(Unit* victim, SpellInfo const* spellProto) const;
+        float  SpellHealingPctDone(Unit* victim, SpellInfo const* spellProto) const;
```
Comment (jackpoz) [inline src/server/game/Entities/Unit/Unit.h]: remove the new space

File: src/server/game/Entities/Unit/Unit.h [hunk 13]
```diff
@@ -1552,7 +1552,7 @@ class TC_GAME_API Unit : public WorldObject
 
         void AddUnitMovementFlag(uint32 f) { m_movementInfo.flags |= f; }
         void RemoveUnitMovementFlag(uint32 f) { m_movementInfo.flags &= ~f; }
-        bool HasUnitMovementFlag(uint32 f) const { return (m_movementInfo.flags & f) == f; }
+        bool   HasUnitMovementFlag(uint32 f) const { return (m_movementInfo.flags & f) == f; }
```
Comment (jackpoz) [inline src/server/game/Entities/Unit/Unit.h]: remove the new space

File: src/server/game/Groups/GroupMgr.h (hunk 8 above)
Comment (jackpoz) [reply src/server/game/Groups/GroupMgr.h]: This still needs to be fixed

File: src/server/game/Loot/LootMgr.h [hunk 14]
```diff
@@ -34,15 +34,15 @@ struct LootItem;
...
-    bool    needs_quest;                                    // quest drop (quest is required for item to drop)
-    uint8   groupid;
-    uint8   mincount;                                       // mincount for drop items
-    uint8   maxcount;                                       // max drop count for the item mincount or Ref multiplicator
-    ConditionContainer conditions;                               // additional loot condition
+    uint32 itemid;                                         // id of the item
+    uint32 reference;                                      // referenced TemplateleId
+    float chance;                                         // chance to drop for both quest and non-quest items, chance to be used for refs
+    uint16 lootmode;
+    bool needs_quest;                                    // quest drop (quest is required for item to drop)
```
Comment (jackpoz) [inline src/server/game/Loot/LootMgr.h]: should you align the comments ?

File: src/server/game/Quests/QuestDef.h [hunk 15]
```diff
@@ -259,7 +259,7 @@ class TC_GAME_API Quest
         std::string const& GetCompletedText() const { return _completedText; }
         int32  GetRewOrReqMoney() const;
         uint32 GetRewHonorAddition() const { return _rewardHonor; }
-        float GetRewHonorMultiplier() const { return _rewardKillHonor; }
+        float  GetRewHonorMultiplier() const { return _rewardKillHonor; }
```
Comment (jackpoz) [inline src/server/game/Quests/QuestDef.h]: remove the new space

**Review by jackpoz — CHANGES_REQUESTED** [review]
Some new spaces have been added, please remove them.

File: src/server/game/Chat/Chat.h [hunk 16]
```diff
@@ -114,7 +114,7 @@ class TC_GAME_API ChatHandler
         char*     extractKeyFromLink(char* text, char const* const* linkTypes, int* found_idx, char** something1 = nullptr);
 
         // if args have single value then it return in arg2 and arg1 == nullptr
-        void      extractOptFirstArg(char* args, char** arg1, char** arg2);
+        void extractOptFirstArg(char* args, char** arg1, char** arg2);
```
Comment (jackpoz) [inline src/server/game/Chat/Chat.h]: could you fix the alignment of the whole ChatHandler class while at it ?

File: src/server/game/Entities/Object/Object.h [hunk 17]
```diff
@@ -251,12 +251,12 @@ class FlaggedValuesArray32
 
         T_FLAGS  GetFlags() const { return m_flags; }
         bool     HasFlag(FLAG_TYPE flag) const { return m_flags & (1 << flag); }
-        void     AddFlag(FLAG_TYPE flag) { m_flags |= (1 << flag); }
-        void     DelFlag(FLAG_TYPE flag) { m_flags &= ~(1 << flag); }
+        void AddFlag(FLAG_TYPE flag) { m_flags |= (1 << flag); }
```
Comment (jackpoz) [inline src/server/game/Entities/Object/Object.h]: Could you fix the alignment of the whole FlaggedValuesArray32 class while at it ?

**Comment by jackpoz** [comment]
I would like to have 1 script that checks all the source code by itself and optionally fixes it automatically

**Comment by Aokromes** [comment]
rgrep "\b  \b" src/* |grep -v "//"
can help to find double spaces on rows, ignoring the ones containing //

File: src/server/game/Battlegrounds/ArenaTeam.h [hunk 18]
```diff
@@ -188,15 +188,15 @@ class TC_GAME_API ArenaTeam
...
 
         uint32 BackgroundColor; // ARGB format
-        uint8  EmblemStyle;     // icon id
+        uint8 EmblemStyle;      // icon id
         uint32 EmblemColor;     // ARGB format
-        uint8  BorderStyle;     // border image id
+        uint8 BorderStyle;      // border image id
         uint32 BorderColor;     // ARGB format
 
         MemberList     Members;
```
Comment (jackpoz) [inline src/server/game/Battlegrounds/ArenaTeam.h]: please remove whitespace from all the members of this class, starting from https://github.com/TrinityCore/TrinityCore/pull/23519/files#diff-d7bafcb23b1ab0c2d6dffc8cd7a74d87R116

**Pull request merged**